x-pack/filebeat/input/awss3/state.go

```diff
@@ -0,0 +1,72 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package awss3
+
+import (
+	"fmt"
+	"time"
+)
+
+// state is used to communicate the publishing state of a s3 object
+type state struct {
```

Comment:
To what extent will modifying this struct affect previously stored states? If changing this can break users deployments (lost state, etc) then I think it's a good idea to document this as a warning to devs. Not sure how hard it would be, but adding a test that decodes a binary blob would help catch cases where a dev modified the struct in a non-backward compatible manner.

Reply:
I added a godoc comment to clarify this aspect. I will think about the unit test